commands/user/events/events.go

```diff
@@ -0,0 +1,73 @@
+package events
+
+import (
+	"fmt"
+
+	"github.com/profclems/glab/commands/cmdutils"
+	"github.com/profclems/glab/internal/utils"
+	"github.com/profclems/glab/pkg/api"
+	"github.com/profclems/glab/pkg/tableprinter"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+)
+
+func NewCmdEvents(f *cmdutils.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "events",
+		Short: "View user events",
+		Args:  cobra.MaximumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			apiClient, err := f.HttpClient()
+			if err != nil {
+				return err
+			}
+
+			events, err := api.CurrentUserEvents(apiClient)
+			if err != nil {
+				return err
+			}
+			projects := make(map[int]*gitlab.Project)
+			for _, e := range events {
+				project, err := api.GetProject(apiClient, e.ProjectID)
+				if err != nil {
+					return err
+				}
+				projects[e.ProjectID] = project
+			}
+
+			title := utils.NewListTitle("User events")
+			title.CurrentPageTotal = len(events)
+
+			if err = f.IO.StartPager(); err != nil {
+				return err
+			}
+			defer f.IO.StopPager()
+			fmt.Fprintf(f.IO.StdOut, "%s\n%s\n", title.Describe(), DisplayAllEvents(events, projects))
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func DisplayAllEvents(events []*gitlab.ContributionEvent, projects map[int]*gitlab.Project) string {
+	table := tableprinter.NewTablePrinter()
+	for _, e := range events {
```

Comment:
How about we display it this way:
```go
for _, e := range events {
   if e.ActionName == "pushed to" {
       fmt.Fprintf(f.IO.StdOut, "Pushed to %s %s at %s\n%q\n", e.PushData.RefType, e.PushData.Ref, projects[e.ProjectID].Name, e.PushData.CommitTitle)
   } else if e.ActionName == "commented on" {
       fmt.Fprintf(f.IO.StdOut, "Commented on %s #%s at %s\n%q\n", e.Note.NoteableType, e.Note.NoteableType, projects[e.ProjectID].Name, e.Note.Body)
   } else {
       fmt.Fprintf(f.IO.StdOut, "%s %q", e.TargetType, e.Title)
   }
    fmt.Fprintln(f.IO.StdOut)  // to leave a blank line 
}
```
The above will display in this format:
```sh
Pushed to branch trunk at NAMESPACE/PROJECT
"chore(deps): bump github.com/xanzy/go-gitlab from 0.38.2 to 0.39.0 ..."

Commented on merge request #305 at NAMESPACE/PROJECT
"CommentBody"

...
```
I haven't tested this out but I think it should be fine. 
We can go a long way to customize output for the other action types just like I have done for `commented on` and `pushed to`.

The supported action types are shown here https://docs.gitlab.com/ee/api/events.html#action-types

Reply:
So I implemented all the actions that I came across. 

I was expecting some of the things to be different like action type `reopened` really was an `e.ActionType = 'reopened'`, but it's just an `opened` Action Type. It's also the same in the Gitlab user contribution feed.

Reply:
Yeah I just noticed that too.

Reply:
I somehow missed checking wiki page creation. Added some special handling for it too.